airbyte-integrations/connectors/source-smartsheets/source_smartsheets/source.py

```diff
@@ -0,0 +1,153 @@
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+"""
+
+import json
+from datetime import datetime
+from typing import Dict, Generator
+
+import smartsheet
+from airbyte_protocol import (
+    AirbyteCatalog,
+    AirbyteConnectionStatus,
+    AirbyteMessage,
+    AirbyteRecordMessage,
+    AirbyteStream,
+    ConfiguredAirbyteCatalog,
+    Status,
+    Type,
+)
+from base_python import AirbyteLogger, Source
+from smartsheet.smartsheet import Smartsheet
+
+
+# helpers
+def get_prop(col_type: str) -> Dict[str, any]:
+    props = {
+        "TEXT_NUMBER": {"type": "string"},
+        "DATE": {"type": "string", "format": "date"},
+        "DATETIME": {"type": "string", "format": "date-time"},
+    }
+    if col_type in props.keys():
+        return props[col_type]
+    else:  # assume string
+        return props["TEXT_NUMBER"]
+
+
+def get_json_schema(sheet: Smartsheet) -> Dict:
+    column_info = {i["title"]: get_prop(i["type"]) for i in sheet["columns"]}
+    json_schema = {
+        "$schema": "http://json-schema.org/draft-07/schema#",
+        "type": "object",
+        "properties": column_info,
+    }
+    return json_schema
+
+
+# main class definition
+class SourceSmartsheets(Source):
+    def check(self, logger: AirbyteLogger, config: json) -> AirbyteConnectionStatus:
+        try:
+            access_token = config["access_token"]
+            spreadsheet_id = config["spreadsheet_id"] + 1
+
+            smartsheet_client = smartsheet.Smartsheet(access_token)
+            smartsheet_client.errors_as_exceptions(True)
+            smartsheet_client.Sheets.get_sheet(spreadsheet_id)
+
+            return AirbyteConnectionStatus(status=Status.SUCCEEDED)
+        except Exception as e:
+            if isinstance(e, smartsheet.exceptions.ApiError):
+                err = e.error.result
+                code = 404 if err.code == 1006 else err.code
+                reason = f"{err.name}: {code} - {err.message} | Check your spreadsheet ID."
+            else:
+                reason = str(e)
+            logger.error(reason)
+        return AirbyteConnectionStatus(status=Status.FAILED)
+
+    def discover(self, logger: AirbyteLogger, config: json) -> AirbyteCatalog:
+        access_token = config["access_token"]
+        spreadsheet_id = config["spreadsheet_id"]
+        streams = []
+
+        smartsheet_client = smartsheet.Smartsheet(access_token)
+        try:
+            sheet = smartsheet_client.Sheets.get_sheet(spreadsheet_id)
+            sheet = json.loads(str(sheet))  # make it subscriptable
+            sheet_json_schema = get_json_schema(sheet)
```

Comment:
`sheet` is a `Dict` not a `Smartsheet` object which is different than the method signature for `get_json_schema` -- should the signature change?

Reply:
you're right. Fixed the type hinting in the arg definition of `get_json_schema`